Wrappers/Python/cil/plugins/astra/utilities/convert_geometry_to_astra_vec_3D.py

```diff
@@ -100,33 +106,63 @@ def convert_geometry_to_astra_vec_3D(volume_geometry, sinogram_geometry_in):
         if sinogram_geometry.geom_type == 'parallel':
             src = system.ray.direction.reshape(3,1)
             projector = 'parallel3d_vec'
-        else:
+        elif sinogram_geometry.geom_type != 'cone_flex':
             src = system.source.position.reshape(3,1)
             projector = 'cone_vec'
+    # Use the per-projection geometry
+    else:
+        # roi 
+        current_centre = [volume_geometry_temp.center_x, volume_geometry_temp.center_y, volume_geometry_temp.center_z]
+
+        # Compute a translation vector that will modify the centre of the reconstructed volume
+        translation = np.array(system.volume_centre.position)
```

Comment:
it should be already a numpy array and it should not require redefinition/cast

Reply:
fixed